File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/MedicinalProductIngredient.ts
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Ratio } from "./Ratio";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type MedicinalProductIngredientSubstance = BackboneElement & {
    code: CodeableConcept;
    strength?: Reference[];
};

export type MedicinalProductIngredientSpecifiedSubstanceStrengthReferenceStrength = BackboneElement & {
    country?: CodeableConcept[];
    strength: Ratio;
    substance?: CodeableConcept;
    measurementPoint?: string;
    strengthLowLimit?: Ratio;
};

export type MedicinalProductIngredientSpecifiedSubstanceStrength = BackboneElement & {
    country?: CodeableConcept[];
    presentation: Ratio;
    concentration?: Ratio;
    measurementPoint?: string;
    referenceStrength?: MedicinalProductIngredientSpecifiedSubstanceStrengthReferenceStrength[];
    presentationLowLimit?: Ratio;
    concentrationLowLimit?: Ratio;
};

export type MedicinalProductIngredientSpecifiedSubstance = BackboneElement & {
    code: CodeableConcept;
    group: CodeableConcept;
    strength?: MedicinalProductIngredientSpecifiedSubstanceStrength[];
    confidentiality?: CodeableConcept;
};

export type MedicinalProductIngredient = DomainResource & {
    role: CodeableConcept;
    substance?: MedicinalProductIngredientSubstance;
    identifier?: Identifier;
    manufacturer?: Reference[];
    specifiedSubstance?: MedicinalProductIngredientSpecifiedSubstance[];
    allergenicIndicator?: boolean;
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/MedicinalProductIngredient.py
from typing import Optional, List
from base import CodeableConcept
from base import DomainResource
from base import Ratio
from base import Reference
from base import Identifier
from base import BackboneElement

class MedicinalProductIngredient_Substance(BackboneElement):
    code: CodeableConcept
    strength: Optional[List[Reference]] = None

class MedicinalProductIngredient_SpecifiedSubstance_Strength_ReferenceStrength(BackboneElement):
    country: Optional[List[CodeableConcept]] = None
    strength: Ratio
    substance: Optional[CodeableConcept] = None
    measurement_point: Optional[str] = None
    strength_low_limit: Optional[Ratio] = None

class MedicinalProductIngredient_SpecifiedSubstance_Strength(BackboneElement):
    country: Optional[List[CodeableConcept]] = None
    presentation: Ratio
    concentration: Optional[Ratio] = None
    measurement_point: Optional[str] = None
    reference_strength: Optional[List[MedicinalProductIngredient_SpecifiedSubstance_Strength_ReferenceStrength]] = None
    presentation_low_limit: Optional[Ratio] = None
    concentration_low_limit: Optional[Ratio] = None

class MedicinalProductIngredient_SpecifiedSubstance(BackboneElement):
    code: CodeableConcept
    group: CodeableConcept
    strength: Optional[List[MedicinalProductIngredient_SpecifiedSubstance_Strength]] = None
    confidentiality: Optional[CodeableConcept] = None

class MedicinalProductIngredient(DomainResource):
    role: CodeableConcept
    substance: Optional[MedicinalProductIngredient_Substance] = None
    identifier: Optional[Identifier] = None
    manufacturer: Optional[List[Reference]] = None
    specified_substance: Optional[List[MedicinalProductIngredient_SpecifiedSubstance]] = None
    allergenic_indicator: Optional[bool] = None